http-client/src/test/groovy/io/micronaut/http/client/SslSpec.groovy

```diff
@@ -34,4 +52,59 @@ class SslSpec extends Specification {
         ctx.close()
         client.close()
     }
+
+    void 'test client ssl handshakeTimeout'() {
+        given:
+        def group = new NioEventLoopGroup()
+        ServerChannel channel = (ServerChannel) new ServerBootstrap()
+                .group(group)
+                .channel(NioServerSocketChannel)
+                .childHandler(new IgnoringChannelInitializer())
+                .bind(0).sync().channel()
+        ApplicationContext cont = ApplicationContext.run(['micronaut.http.client.ssl.handshakeTimeout': '1s'])
+        def address = (InetSocketAddress) channel.localAddress()
+        when:
+        Flux.from(cont.getBean(HttpClient).exchange("https://localhost:$address.port"))
+                .blockFirst(Duration.ofSeconds(5))
+        then:
+        def e = thrown RuntimeException
+        e.cause instanceof SslHandshakeTimeoutException
+        cleanup:
+        group.shutdownGracefully()
+    }
+
+    void 'test server ssl handshakeTimeout'() {
+        given:
+        def group = new NioEventLoopGroup()
+        ApplicationContext context = ApplicationContext.run([
+                'micronaut.ssl.port'            : -1,
+                'micronaut.ssl.enabled'         : true,
+                'micronaut.ssl.buildSelfSigned' : true,
+                'micronaut.ssl.handshakeTimeout': '1s',
+        ])
+        def embeddedServer = context.getBean(EmbeddedServer)
+        embeddedServer.start()
+        def channel = new Bootstrap()
+                .channel(NioSocketChannel)
+                .group(group)
+                .handler(new IgnoringChannelInitializer())
+                .connect(embeddedServer.host, embeddedServer.port).sync().channel()
+        expect:
+        // returns false on timeout
+        channel.closeFuture().await(5, TimeUnit.SECONDS)
```

Comment:
This is testing the timeout failure right? Wouldn't this test fail if it returned false?

Reply:
The channel should be closed after one second by the server because of the server timeout, and then the await on the client shouldn't timeout

Reply:
I see, thanks